### PR TITLE
Don't increment fingerprint change count if assessment is closed

### LINKED ISSUE #9588: Don't increment fingerprint count when assessment is closed
Fingerprint change count is useful to detect if a student is accessing an assessment (particularly an exam) from more than one device. However, after it's closed, students may access the assessment outside of the controlled environment without problems. At that point, the fingerprint change is still incremented. This means that my window for checking fingerprint counts is limited to from when the exam ends and when I release it for viewing. In my case this is minor, since this time is non-trivial, but in cases where the exam has limited manual grading, it can be released right away, so investigating the count would be hard. So, I think it's better if closed assessments do not trigger a fingerprint count.

This may also be extended to fingerprint changes while the access rules have active set to false.

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Resolves #9588. I also disabled the count when active is false, mostly to handle cases where the assessment is not actually closed but is closed in effect through an active: false access rule; though I am open to discussion if that is a valid approach.

### CODE REVIEW

**Comment by nwalters512** [comment]
I don't love the fact that instructors will seemingly inexplicably get different results if they manually compute the change count vs. just looking at the count we show them. If we do really want this change (I need a little more convincing that we do), we should explain this in the documentation and ideally in the product. We'll also need to explain that the count changed meaning at a certain date/time; I'm not sure how best to do that. We could also consider an "active" change count separately?

I am not yet requesting any specific changes, just raising some points for discussion.

**Comment by jonatanschroeder** [comment]
> I don't love the fact that instructors will seemingly inexplicably get different results if they manually compute the change count vs. just looking at the count we show them. If we do really want this change (I need a little more convincing that we do), we should explain this in the documentation and ideally in the product. We'll also need to explain that the count changed meaning at a certain date/time; I'm not sure how best to do that. We could also consider tracking an "active" change count separately?
> 
> I am not yet requesting any specific changes, just raising some points for discussion.

The main reason that this field exists (as described in the helper texts where the value is shown) is to support instances of academic integrity checks. Including accesses that happen while the assessment is already closed defeats this purpose. As an instructor I don't care where the student reviews the exam after it is closed (assuming I have given them access to it), I'm interested in accesses while submissions are allowed.

I added some comments in the helper texts to make this clearer. Instructors should be able to count by basically checking the values up until the "Close" event.

**Review by nwalters512 — APPROVED** [review]
I'd appreciate it if you could update `docs/assessment.md` to reflect the new behavior.

File: docs/assessment.md [hunk 1]
```diff
@@ -544,6 +544,8 @@ The appropriate link to provide to students can be found by opening the "Setting
...
 
-Each time the client fingerprint is recorded, PrairieLearn checks it against the previous client fingerprint used to access that assessment. PrairieLearn will track and display how many times the fingerprint has changed during the course of the assessment. You can see the number of times a fingerprint changed on the Students tab of an assessment. You can also see more detailed information about the client fingerprint by accessing the student's assessment instance and reviewing the Fingerprint column of the event log.
+While an assessment is open, each time the client fingerprint is recorded, PrairieLearn checks it against the previous client fingerprint used to access that assessment. PrairieLearn will track and display how many times the fingerprint has changed during the course of the assessment. You can see the number of times a fingerprint changed on the Students tab of an assessment. You can also see more detailed information about the client fingerprint by accessing the student's assessment instance and reviewing the Fingerprint column of the event log.
```
Comment (nwalters512) [inline docs/assessment.md:547]:
```suggestion
While an assessment is open and active, each time the client fingerprint is recorded, PrairieLearn checks it against the previous client fingerprint used to access that assessment. PrairieLearn will track and display how many times the fingerprint has changed during the course of the assessment. You can see the number of times a fingerprint changed on the Students tab of an assessment. You can also see more detailed information about the client fingerprint by accessing the student's assessment instance and reviewing the Fingerprint column of the event log.
```

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**